src/main/java/org/opentripplanner/middleware/itinerarymatching/ItineraryChecker.java

```diff
@@ -0,0 +1,50 @@
+package org.opentripplanner.middleware.itinerarymatching;
+
+import org.opentripplanner.middleware.otp.LegFinder;
+import org.opentripplanner.middleware.otp.response.Itinerary;
+import org.opentripplanner.middleware.otp.response.Leg;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.opentripplanner.middleware.itinerarymatching.ItineraryFromLegMatcher.getTransitLegs;
+
+/**
+ * Helper class for performing an itinerary existence check.
+ */
+public class ItineraryChecker {
+
+    private final Itinerary itinerary;
+
+    private final LegFinder legFinder;
+
+    private final LocalDate targetDate;
+
+    public ItineraryChecker(Itinerary itinerary, LegFinder legFinder, LocalDate targetDate) {
+        this.itinerary = itinerary;
+        this.legFinder = legFinder;
+        this.targetDate = targetDate;
+    }
+    /**
+     * Check leg existence and returns status and delay information
+     */
+    public ItineraryCheckStatus checkLegs() {
+        List<Leg> transitLegs = getTransitLegs(itinerary.legs);
```

Comment:
I think `itinerary.legs` could be null at this point?

Reply:
See https://github.com/ibi-group/otp-middleware/pull/357/commits/bc9312fd7e1e57fe19b3a0a4bc2c15020903895c